src/services/message-formating/message-formatting.service.ts

```diff
@@ -0,0 +1,40 @@
+import _ from "lodash"
+import { TextFormats } from '../../enums/text-format.enum'
+import { BoldererService } from './bolderer/bolderer.service'
+
+export class MessageFormattingService {
+  private static _instance: MessageFormattingService
+
+  public static getInstance(): MessageFormattingService {
+    if (_.isNil(MessageFormattingService._instance)) {
+      MessageFormattingService._instance = new MessageFormattingService()
+    }
+    return MessageFormattingService._instance
+  }
+
+  public format(formatToApply: string, text: string) {
+    switch (formatToApply) {
+      case TextFormats.BOLD: {
+        text = BoldererService.getInstance().bold(text)
```

Comment:
If really think that:
- switch are bad by design and if/else if should be used instead
- you have a great opportunity here to simply this process with a map

To create a map:
- Change the values of the enum to match the names of the methods
- Instead of having a feature per text type, create a generic service
- Create each method matching the value you set earlier on the enum
- Simply do `return YourService.getInstance()[formatToApply](text)`

Reply:
Ok, I will:
- Use a if/else if conditions.
- Change the name of the enum
- Create a method per format directly in the message formatter service.